xbmc/network/Socket.h

```diff
@@ -70,21 +75,56 @@ namespace SOCKETS
 
     void SetAddress(const char *address)
     {
+      in6_addr addr6;
       memset(&saddr, 0, sizeof(saddr));
-      saddr.sin_family = AF_INET;
-      saddr.sin_addr.s_addr = inet_addr(address);
-      size = sizeof(saddr);
+      if (inet_pton(PF_INET6, address, &addr6) == 1)
```

Comment:
should be `AF_INET6` and why not use inet_pton for ipv4 as well? would make more sense to stick to one method.
Also there's no sanity checking here, it just assumes that not ipv6 is ipv4 which isn't true, it could very well be that the string is a bad ipv6 address.